internal/v5/resources.go

```diff
@@ -18,6 +18,15 @@ import (
 // GET id/meta/resources
 // https://github.com/juju/charmstore/blob/v5/docs/API.md#get-idmetaresources
 func (h *ReqHandler) metaResources(entity *mongodoc.Entity, id *router.ResolvedURL, path string, flags url.Values, req *http.Request) (interface{}, error) {
+	if entity.URL.Series == "bundle" {
+		// Bundles do not have resources so we return an empty result.
+		return []params.Resource{}, nil
+	}
+	if entity.CharmMeta == nil {
```

Comment:
Why not just change these checks in basicListResources instead?
(also, if CharmMeta isn't set, something is seriously wrong or the meta endpoint isn't configured right, either of which would justify a panic, I think)